src/core/qgsvectorfilewriter.cpp

```diff
@@ -2253,6 +2263,29 @@ gdal::ogr_feature_unique_ptr QgsVectorFileWriter::createFeature( const QgsFeatur
 
       case QVariant::Invalid:
         break;
+
+      case QVariant::List:
+        // only string list supported at the moment, fall through to default for other types
+        if ( field.subType() == QVariant::String )
+        {
+          QStringList list = attrValue.toStringList();
+          int count = list.count();
+          char **lst = new char *[count];
+          if ( count > 0 )
+          {
+            int pos = 0;
+            for ( QString string : attrValue.toStringList() )
+            {
+              lst[pos] = mCodec->fromUnicode( string ).data();
+              pos++;
+            }
+          }
+          OGR_F_SetFieldStringList( poFeature.get(), ogrField, lst );
```

Comment:
https://www.gdal.org/ogr__api_8h.html#a35dd76723ae1acc91135054e64d7598e => lst should have last element as null ptr as in https://github.com/qgis/QGIS/blob/04c8f5b6f9a4fae67bfc9889ba120f91dcb93193/src/core/qgsvectorfilewriter.cpp#L309

Reply:
I shall do.